src/parser/jobs/blm/modules/Thunder.js

```diff
@@ -0,0 +1,241 @@
+
+		// Make sure we're tracking for this target
+		const applicationKey = `${event.targetID}|${event.targetInstance}`
+		const lastApplication = this._lastApplication[applicationKey] = this._lastApplication[applicationKey] || {}
+
+		// If it's not been applied yet, set it and skip out
+		if (!lastApplication[statusId]) {
+			lastApplication[statusId] = event.timestamp
+			//save the application for later use in the output
+			this._pushApplication(applicationKey, statusId, event, null)
+			return
+		}
+		// Base clip calc
+		let clip = STATUS_DURATION[statusId] - (event.timestamp - lastApplication[statusId])
+		clip = Math.max(0, clip)
+		// Capping clip at 0 - less than that is downtime, which is handled by the checklist requirement
+		this._clip[statusId] += clip
+		//save the application for later use in the output
+		this._pushApplication(applicationKey, statusId, event, clip)
+
+		lastApplication[statusId] = event.timestamp
+	}
+
+	// Get the uptime percentage for the Thunder status debuff
+	getThunderUptime() {
+		const statusTime = this.entityStatuses.getStatusUptime(STATUSES.THUNDER_III.id, this.enemies.getEntities())
+		const uptime = this.parser.fightDuration - this.invuln.getInvulnerableUptime()
+		return (statusTime / uptime) * 100
+	}
+
+	_onComplete() {
+		// Checklist item for keeping Thunder 3 DoT rolling
+		this.checklist.add(new Rule({
+			name: <Trans id="blm.thunder.checklist.dots.name">Keep your <StatusLink {...STATUSES.THUNDER_III} /> DoT up</Trans>,
+			description: <Trans id="blm.thunder.checklist.dots.description">
+				Your <StatusLink {...STATUSES.THUNDER_III} /> DoT contributes significantly to your overall damage, both on its own, and from additional <StatusLink {...STATUSES.THUNDERCLOUD} /> procs. Try to keep the DoT applied.
+			</Trans>,
+			target: 95,
+			requirements: [
+				new Requirement({
+					name: <Trans id="blm.thunder.checklist.dots.requirement.name"><StatusLink {...STATUSES.THUNDER_III} /> uptime</Trans>,
+					percent: () => this.getThunderUptime(),
+				}),
+			],
+		}))
+
+		// Suggestions to not spam T3 too much
+		const sumClip = this._clip[STATUSES.THUNDER_III.id]
+		const maxExpectedClip = (this.thunder3Casts - 1) * MAX_ALLOWED_T3_CLIPPING
+		if (sumClip > maxExpectedClip) {
+			this.suggestions.add(new Suggestion({
+				icon: ACTIONS.THUNDER_III.icon,
+				content: <Trans id="blm.thunder.suggestions.excess-thunder.content">
+					Casting <ActionLink {...ACTIONS.THUNDER_III} /> too frequently can cause you to lose DPS by casting fewer <ActionLink {...ACTIONS.FIRE_IV} />. Try not to cast <ActionLink showIcon={false} {...ACTIONS.THUNDER_III} /> unless your <StatusLink {...STATUSES.THUNDER_III} /> DoT or <StatusLink {...STATUSES.THUNDERCLOUD} /> proc are about to wear off.
+					Check the <a href="javascript:void(0);" onClick={() => this.parser.scrollTo(this.constructor.handle)}><NormalisedMessage message={this.constructor.title}/></a> module for more information.
```

Comment:
React is emitting deprecation errors for href="javascript:void(0)" links: `Warning: A future version of React will block javascript: URLs as a security precaution. Use event handlers instead if you can. If you need to generate unsafe HTML try using dangerouslySetInnerHTML instead. React was passed "javascript:void(0);"`

The better way to do this would be:
```suggestion
					Check the <a href="#" onClick={e => { e.preventDefault(); this.parser.scrollTo(this.constructor.handle) }}><NormalisedMessage message={this.constructor.title}/></a> module for more information.
```

Reply:
alrighty, I dunno enough about links so I'll stick with your suggestion